spinalcordtoolbox/image.py

```diff
@@ -503,53 +503,44 @@ def save(self, path=None, dtype=None, verbose=1, mutable=False):
 
         :param mutable: whether to update members with newly created path or dtype
         """
+        if not mutable:
+            self.copy().save(path, dtype, verbose, mutable=True)
+            return self
```

Comment:
This is a really interesting pattern! It took me a minute to wrap my head around, though, so I had one small tweak that might make it a bit easier to grok?

```suggestion
        if mutable:
            pass  # changes to the path/dtype/header will be applied in-place
        else:
            # otherwise, we work with a copy (to preserve the state of the original image object)
            self.copy().save(path, dtype, verbose, mutable=True)
            return self
```

I mainly suggest this because `mutable` is off by default, and will only ever be specified via `mutable=True`. So, I find it helpful to clarify the positive case (`if mutable`) rather than the negative case (`if not mutable`).

Reply:
Sure, if there's an `else` branch it's generally good to make the `if` condition "positive". Instead of `pass` though, we might as well put the rest of the code there. I'll make the change.

Reply:
Done!

Reply:
> Instead of pass though, we might as well put the rest of the code there. I'll make the change.

My only concern with this is readability -- with big if/else blocks, I find it a little cumbersome to have to scroll all the way to the bottom to see what the `else:` is... I like being able to read both if/else comments in one small, sequential set of lines?

That said, taking a look at your changes, the `if mutable` block does all fit into one 1080p screen, so... maybe it's not too bad?